packages/ui-react-utils/src/omitProps.ts

```diff
@@ -26,34 +26,35 @@
  * ---
  * category: utilities/react
  * ---
- * Return an object with the remaining props after propTypes (and additionally specified props) omitted.
+ * Return an object with the remaining props after the given props are omitted.
  *
- * Automatically excludes the following props: 'theme', 'children', 'className', 'style', 'styles', 'makeStyles', 'themeOverride'
+ * Automatically excludes the following props:
+ * 'theme', 'children', 'className', 'style', 'styles', 'makeStyles', 'themeOverride'
  * @module omitProps
- * @param {Object} props React component props
- * @param {Object|Array<string>} propTypes React component propTypes or the list of allowed prop keys
- * @param {Array} exclude an optional array of prop names to exclude
- * @returns {Object} props object without the excluded props
+ * @param props The object to process
+ * @param propsToOmit list disallowed prop keys or an object whose
+ *        keys will be omitted.
+ * @param exclude an optional array of disallowed prop names to omit
+ * @returns props object without the excluded props
  * @module omitProps
  */
 function omitProps<T extends Record<string, any>>(
   props: T,
-  propTypesOrAllowedPropList: Record<string, any> | string[],
+  propsToOmit: Record<string, any> | string[],
   exclude?: string[]
 ) {
-  const propKeys = Array.isArray(propTypesOrAllowedPropList)
-    ? propTypesOrAllowedPropList
-    : Object.keys(propTypesOrAllowedPropList || {})
-  const combined = exclude ? propKeys.concat(exclude) : propKeys
-
-  return omit(props, combined)
+  const propKeys = Array.isArray(propsToOmit)
+    ? propsToOmit
+    : Object.keys(propsToOmit || {})
+  const keysToOmit = exclude ? propKeys.concat(exclude) : propKeys
+  return omit(props, keysToOmit) as Record<string, any>
```

Comment:
While it returns `Partial<P>` this causes lots of issues since it will try to put conflicting props where its used.

Reply:
I'd leave a TODO here to try to type it later once all props are in place